### PR TITLE
Frontend tests

### PR DESCRIPTION
**Pull request by nbarlowATI**
Add some basic tests for the Flask frontend.

### CODE REVIEW

File: dtbase/tests/test_frontend_locations.py [hunk 1]
```diff
@@ -0,0 +1,27 @@
...
+        assert response.status_code == 200
+        html_content = response.data.decode("utf-8")
+        assert "Backend API not found" in html_content
+
+
+def test_new_location_schema(frontend_client):
+    with frontend_client:
+        with requests_mock.Mocker() as m:
+            m.get("http://localhost:5000/location/list_location_schemas", json=[])
+            m.get("http://localhost:5000/location/list_location_identifiers", json=[])
```
Comment (mastoffel) [inline dtbase/tests/test_frontend_locations.py:23]: Just for me to learn: Why are you doing these get requests?

File: dtbase/webapp/app/base/templates/errors/backend_not_found.html
Comment (mastoffel) [inline dtbase/webapp/app/base/templates/errors/backend_not_found.html:1]: Nice! Good looking 404 page.

File: dtbase/webapp/app/models/routes.py
Comment (mastoffel) [inline dtbase/webapp/app/models/routes.py:1]: The models frontend doesn't work for me (again a json.loads thing - when I change that in the backend  model/routes.py it works).

**Review by mastoffel — COMMENTED** [review]
Great job! Lots of work in here. The tests all run for me and make sense. A few comments below, but I think the models thing can be addressed later.

File: dtbase/tests/test_frontend_locations.py (hunk 1 above)
Comment (nbarlowATI) [reply dtbase/tests/test_frontend_locations.py:23]: These are "mock" get requests.   With this, if anything in the code ends up sending a "get" request to those urls, it won't actually send the request, but instead the mocker will return whatever is in the "json" argument (in this case, empty lists).    
This is a way of testing the frontend, without needing to have the backend running...

**Pull request merged**